### PR TITLE
Add b:lion_squeeze_spaces

### PR DESCRIPTION
**Pull request by sudo-nice**
This PR is to fix https://github.com/tommcdo/vim-lion/issues/15

### CODE REVIEW

File: plugin/lion.vim [hunk 1]
```diff
@@ -2,6 +2,8 @@
 " Maintainer:   Tom McDonald <http://github.com/tommcdo>
 " Version:      1.0
 
+let b:lion_squeeze_spaces = get(b:, 'lion_squeeze_spaces', 0)
```
Comment (tommcdo) [inline plugin/lion.vim]: We can't rely on one-time initialization for a buffer-local variable, as a new instance of it is created for each buffer (which could happen long after Vim initialization). Instead, we should use this `get()` call in the `if`-statement.

I had an idea to check the buffer variable first, then fall back to the global variable, then to a default value. What do you think?

**Comment by sudo-nice** [comment]
Please, check the new commit.

File: plugin/lion.vim [hunk 2]
```diff
@@ -62,6 +62,22 @@ function! s:align(mode, type, vis, align_char)
 		endif
 		let [start_line, end_line, start_col, end_col, middle_start_col, middle_end_col] = pos
 
+		" Check for 'lion_squeeze_spaces' option
+		if get(b:, 'lion_squeeze_spaces', 0)
```
Comment (tommcdo) [inline plugin/lion.vim]: A likely use case might be to turn on the global option, but disable it for some buffers (perhaps with an auto command for a particular file type). In order to support that, we'd need to distinguish between 0 and unset.

Comment (sudo-nice) [inline plugin/lion.vim]: You're right. What about the last commit?

**Comment by wellle** [comment]
Works great, thank you both! 👍

**Pull request merged**